### PR TITLE
OMEMO Rework

### PR DESCRIPTION
**Pull request by StefanKropp**

### CODE REVIEW

**Comment by jubalh** [comment]
Cool!
I think the `Rework` in each commit isn't necessary. Just state what you changed in subject line :)
And in the message body additionally to what you did (`Moved omemo_devicelist_subscribe after loading OMEMO files`) I would also like to understand *why* you did it :)

**Comment by jubalh** [comment]
@StefanKropp I updated the commit messages. Please pull before pushing.

File: tests/unittests/omemo/stub_omemo.c [hunk 1]
```diff
@@ -114,3 +114,11 @@ omemo_encrypt_file(FILE* in, FILE* out, off_t file_size, int* gcry_res)
     return NULL;
 };
 void omemo_free(void* a){};
+
+uint32_t
+omemo_device_id() {
+	return 123;
+}
+
```
Comment (jubalh) [inline tests/unittests/omemo/stub_omemo.c:122]: why so many empty lines?

**Comment by StefanKropp** [comment]
> Please squash the last two commits into one and replace `SENT` with `SEND`.
> Do we want all this as info? Some looks more like debug?
I recommend we use this branch as a kind of "testing" brunch. In debug mode there are do much information.
When we understood the code and found the problem, we can "clean-up" the code and provide a clear commit for those changes.

**Comment by jubalh** [comment]
Ok, so we will do this before the merge :)

**Comment by jubalh** [comment]
Done again and merged in: https://github.com/profanity-im/profanity/pull/1539

**Pull request closed without merging**